specification/netapp/resource-manager/Microsoft.NetApp/stable/2019-11-01/netapp.json

```diff
@@ -0,0 +1,2699 @@
+        "description": "Patch the specified volume",
+        "parameters": [
+          {
+            "name": "body",
+            "description": "Volume object supplied in the body of the operation.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/volumePatch"
+            }
+          },
+          {
+            "$ref": "#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroup"
+          },
+          {
+            "$ref": "#/parameters/AccountName"
+          },
+          {
+            "$ref": "#/parameters/PoolName"
+          },
+          {
+            "$ref": "#/parameters/VolumeName"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/volume"
+            }
+          },
+          "202": {
+            "description": "Accepted -- Create, update or delete request accepted; operation will complete asynchronously"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed."
+          }
+        },
+        "x-ms-examples": {
+          "Volumes_Update": {
+            "$ref": "examples/Volumes_Update.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Volumes"
+        ],
+        "operationId": "Volumes_Delete",
+        "summary": "Delete a volume",
+        "description": "Delete the specified volume",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroup"
+          },
+          {
+            "$ref": "#/parameters/AccountName"
+          },
+          {
+            "$ref": "#/parameters/PoolName"
+          },
+          {
+            "$ref": "#/parameters/VolumeName"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted -- Create, update or delete request accepted; operation will complete asynchronously"
+          },
+          "204": {
+            "description": "NoContent -- Resource does not exist."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed."
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Volumes_Delete": {
+            "$ref": "examples/Volumes_Delete.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.NetApp/netAppAccounts/{accountName}/capacityPools/{poolName}/volumes/{volumeName}/revert": {
+      "post": {
```

Comment:
>post [](start = 7, length = 4)

you are missing the x-ms-long-running-operation annotation - please take a look at [https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation)

Reply:
Yes. This was also pointed out in the review for the previous swagger which was accepted. We created an open internal ticket on this issue to investigate.

The fact is that adding this results in a failure from the autorest validation 'ERROR (LongRunningResponseStatusCode/R2005/SDKViolation): A 'POST' operation 'Volumes_Revert' with x-ms-long-running-operation extension must have a valid terminal success status code 200 or 201 or 204.'

But perhaps you have some idea how to solve this?

Reply:
Yup. Due to limitations of the current version of OpenAPiSpec (aka Swagger), we don't have a good way of modeling the final response on a long running operation (the response on the GET on the async operation URL).

The, somewhat counterintuitive, workaround is to add a 200 response to this path. By convention the code generator will understand that your service is not returning the 200 response for this POST, but that this is the final operation result instead. Does that make sense?

---
In reply to: [369432464](https://github.com/Azure/azure-rest-api-specs/pull/8164#discussion_r369432464) [](ancestors = 369432464)

Reply:
So I just need to add a (dummy) 200 response block. I'll update.

Reply:
@majastrz  Update made. But I notice some validation checks did not run.